scene/gui/button.cpp

```diff
@@ -73,15 +112,15 @@ void Button::_notification(int p_what) {
 			Color color;
 			Color color_icon(1, 1, 1, 1);
 
-			Ref<StyleBox> style = get_theme_stylebox(SNAME("normal"));
+			Ref<StyleBox> style = theme_cache.normal;
 			bool rtl = is_layout_rtl();
 
 			switch (get_draw_mode()) {
 				case DRAW_NORMAL: {
 					if (rtl && has_theme_stylebox(SNAME("normal_mirrored"))) {
-						style = get_theme_stylebox(SNAME("normal_mirrored"));
+						style = theme_cache.normal_mirrored;
```

Comment:
I see most of this code uses `has_theme_*` checks before fetching the theme items, but the new cache doesn't.
Can that lead to issues?

Reply:
These checks are kind of iffy to begin with, we normally don't code controls like that. Issues this should not create, after all the cache would simply host default values for each data type. But these `has_theme_*` calls are potentially excessive, same as `get_theme_*` calls were, and we aren't using any cache for them. I think we could simply use the cached values, or at least the dumb cache.

But I've decided to get back to this later.